src/apps/bridge/sensor_drivers/seapointTurbiditySensor.cpp

```diff
@@ -0,0 +1,175 @@
+
+        uint32_t current_timestamp = pdTICKS_TO_MS(xTaskGetTickCount());
+        if (current_timestamp - turbidity_sensor->last_timestamp >
+                DEFAULT_TURBIDITY_READING_PERIOD_MS + 1000U ||
+            turbidity_sensor->reading_count == 1U) {
+          printf("Updating Seapoint Turbidity %016" PRIx64
+                 " node position, current_time = %" PRIu32 ", last_time = %" PRIu32
+                 ", reading count: %" PRIu32 "\n",
+                 node_id, current_timestamp, turbidity_sensor->last_timestamp,
+                 turbidity_sensor->reading_count);
+          turbidity_sensor->node_position =
+              topology_sampler_get_node_position(node_id, pdTICKS_TO_MS(5000));
+        }
+        turbidity_sensor->last_timestamp = current_timestamp;
+
+        size_t log_buflen =
+            snprintf(log_buf, SENSOR_LOG_BUF_SIZE,
+                     "%016" PRIx64 ","     // Node Id
+                     "%" PRIi8 ","         // node_position
+                     "seapoint_turbidity," // node_app_name
+                     "%" PRIu64 ","        // reading_uptime_millis
+                     "%" PRIu64 "."        // reading_time_utc_ms seconds part
+                     "%03" PRIu32 ","      // reading_time_utc_ms millis part
+                     "%" PRIu64 "."        // sensor_reading_time_ms seconds part
+                     "%03" PRIu32 ","      // sensor_reading_time_ms millis part
+                     "%.4f,"               // s_signal
+                     "%.3f\n",             // r_signal
+                     node_id, turbidity_sensor->node_position,
+                     turbidity_data.header.reading_uptime_millis, reading_time_sec,
+                     reading_time_millis, sensor_reading_time_sec, sensor_reading_time_millis,
+                     turbidity_data.s_signal, turbidity_data.r_signal);
+        if (log_buflen > 0) {
+          BRIDGE_SENSOR_LOG_PRINTN(BM_COMMON_IND, log_buf, log_buflen);
+        } else {
+          printf("ERROR: Failed to print Seapoint Turbidity individual reading to log\n");
+        }
+        vPortFree(log_buf);
+      }
+      xSemaphoreGive(turbidity_sensor->_mutex);
+    } else {
+      printf(
+          "Failed to take mutex for Seapoint Turbidity Sensor after getting a new reading\n");
+    }
+  }
+}
+
+void SeapointTurbiditySensor::aggregate(void) {
+  char *log_buf = static_cast<char *>(pvPortMalloc(SENSOR_LOG_BUF_SIZE));
+  configASSERT(log_buf);
+  if (xSemaphoreTake(_mutex, portMAX_DELAY)) {
+    size_t log_buflen = 0;
+    seapoint_turbidity_aggregations_t turbidity_aggs = {
+        .turbidity_s_mean_ftu = NAN, .turbidity_r_mean_ftu = NAN, .reading_count = 0};
+    if (turbidity_s_ftu.getNumSamples() > MIN_READINGS_FOR_AGGREGATION) {
+      turbidity_aggs.turbidity_s_mean_ftu = turbidity_s_ftu.getMean();
+      turbidity_aggs.turbidity_r_mean_ftu = turbidity_r_ftu.getMean();
+      turbidity_aggs.reading_count = reading_count;
+    }
```

Comment:
Normally we do some sort of sanity check on here for min/max values, but I talked to Evan about this and we agreed to just let Spotters bitpacking do the filtering based on the encodable min/max values.